src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -848,61 +938,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+        if (!_filteredFonts)
         {
-            if (font.LocalizedName() == currentFont)
-            {
-                result = true;
-            }
+            _updateFilteredFontList();
         }
-        return result;
+        return _filteredFonts;
     }
 
     // Method Description:
     // - Determines whether we should show the list of all the fonts, or we should just show monospace fonts
     bool Appearances::ShowAllFonts() const noexcept
     {
-        // - _ShowAllFonts is directly bound to the checkbox. So this is the user set value.
-        // - If we are not using a monospace font, show all of the fonts so that the ComboBox is still properly bound
-        return _ShowAllFonts || !UsingMonospaceFont();
+        return _ShowAllFonts;
     }
 
-    void Appearances::ShowAllFonts(const bool& value)
+    void Appearances::ShowAllFonts(const bool value)
     {
         if (_ShowAllFonts != value)
         {
             _ShowAllFonts = value;
+            _filteredFonts = nullptr;
             PropertyChanged.raise(*this, PropertyChangedEventArgs{ L"ShowAllFonts" });
+            PropertyChanged.raise(*this, PropertyChangedEventArgs{ L"FilteredFontList" });
         }
     }
 
-    IInspectable Appearances::CurrentFontFace() const
+    void Appearances::FontFaceBox_GotFocus(const Windows::Foundation::IInspectable& sender, const RoutedEventArgs&)
     {
-        const auto& appearanceVM{ Appearance() };
-        const auto appearanceFontFace{ appearanceVM.FontFace() };
-        return box_value(ProfileViewModel::FindFontWithLocalizedName(appearanceFontFace));
+        _updateFontNameFilter({});
+        sender.as<AutoSuggestBox>().IsSuggestionListOpen(true);
     }
 
-    void Appearances::FontFace_SelectionChanged(const IInspectable& /*sender*/, const SelectionChangedEventArgs& e)
+    void Appearances::FontFaceBox_LostFocus(const IInspectable& sender, const RoutedEventArgs&)
     {
-        // NOTE: We need to hook up a selection changed event handler here instead of directly binding to the appearance view model.
-        //       A two way binding to the view model causes an infinite loop because both combo boxes keep fighting over which one's right.
-        const auto selectedItem{ e.AddedItems().GetAt(0) };
-        const auto newFontFace{ unbox_value<Editor::Font>(selectedItem) };
-        Appearance().FontFace(newFontFace.LocalizedName());
-        if (!UsingMonospaceFont())
+        const auto appearance = Appearance();
+        const auto fontSpec = sender.as<AutoSuggestBox>().Text();
+
+        if (fontSpec.empty())
         {
-            ShowProportionalFontWarning(true);
+            appearance.ClearFontFace();
         }
         else
         {
-            ShowProportionalFontWarning(false);
+            appearance.FontFace(fontSpec);
+        }
+
+        // TODO: Any use of FindFontWithLocalizedName is broken and requires refactoring in time for version 1.21.
```

Comment:
on a scale of 1-scary, this comment sounds scary? Like, ship-blocking scary. Do we have a plan for this?

Reply:
Font features and axes have multiple blocking bugs right now (even without this PR), so I figured that we can do a follow-up refactoring later where we fix both issues: The existing bugs and the incompatibilities introduced by this one. I'm half done making the changes to fix this and they're highly related.

Reply:
Mmk. I'll file something and tag you on it just to make sure we don't lose it.